packages/react-components/react-table/src/hooks/useMeasureElement.ts

```diff
@@ -0,0 +1,59 @@
+import { canUseDOM } from '@fluentui/react-utilities';
+import * as React from 'react';
+import { useCallback, useEffect, useRef, useState } from 'react';
+
+/**
+ * This hook does two things:
+ *  - reports width of the element whose ref has been passed as the parameter
+ *  - as the parent resizes, it sets the ref element width to always fill the parent
+ *
+ * It creates a dummy element adjacent to the one we are trying to measure and
+ * uses it to track the available space for the element.
+ *
+ * This approach is also faster than trying to measure the final element (e.g. a table)
+ * which can take much more time to be laid out than an empty div.
+ *
+ * @param el element to measure and report width on
+ * @returns
+ */
+export function useMeasureElement(el: React.RefObject<HTMLElement>) {
```

Comment:
```suggestion
export function useMeasureElement(): React.RefObject<HTMLElement> {
```

Accepting refs as a param in an anti-pattern. If you need to use refs, your reusable hook should return it.

By accepting refs you will never know when a ref will be updated i.e. you cannot handle updates and your handlers will stuck on old element.